### PR TITLE
Introduce hotel ratings Api Support

### PR DESCRIPTION
**Pull request by SealSoft**
#29 Add support for hotel ratings api

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -307,6 +307,8 @@ HotelOffer[] offers = amadeus.shopping.hotelOffers.get(Params
 HotelOffer hotelOffer = amadeus.shopping.hotelOffersByHotel.get(Params.with("hotelId", "BGLONBGB"));
 // Confirm the availability of a specific offer
 HotelOffer offer = amadeus.shopping.hotelOffer("4BA070CE929E135B3268A9F2D0C51E9D4A6CF318BA10485322FA2C7E78C7852E").get();
+// Hotel Ratings
+HotelRating[] hotelRatings = amadeus.shopping.hotelRatings.get(Params.with("hotelIds", "ELONMFS,ADNYCCTB"));
```
[the remark this answers is not in the document]
Comment (SealSoft) [reply README.md]: The code style pattern does not apply to variable name such as `eReputation` so i renamed it as `ereputation`.
Calling the api would be now `amadeus.ereputation.hotelSentiments.get()`

[the remark this answers is not in the document]
Comment (anthonyroux) [reply README.md]: You are right!

[the remark this answers is not in the document]
Comment (anthonyroux) [reply README.md]: Why it does not apply to variables like this?

[the remark this answers is not in the document]
Comment (SealSoft) [reply README.md]: False alarm. My bad. I thought the regex request the first 2 characters as lowercase. Fixed!

[the remark this answers is not in the document]
Comment (SealSoft) [reply README.md]: I ve just commit a version with the variable named `eReputation` and travis ci failed with message: `[ant:checkstyle] [ERROR] /home/travis/build/amadeus4dev/amadeus-java/src/main/java/com/amadeus/Amadeus.java:56:22: Member name 'eReputation' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MemberName]`

[the remark this answers is not in the document]
Comment (SealSoft) [reply README.md]: I will restore it to the `ereputation` version and you can inform me how to handle this issue

[the remark this answers is not in the document]
Comment (anthonyroux) [reply README.md]: Fair enough, we never had this case with one letter. Let's keep it like this then, thanks and sorry for the time you spent on this.

File: src/main/java/com/amadeus/resources/HotelRating.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.amadeus.resources;
+
+import lombok.Getter;
+
+public class HotelRating extends Resource {
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/HotelRating.java]: The class should be named HotelSentiment as it is the object returned by the API:

image

image

I fully agree: this is confusing. We changed the "commercial" name of the API to be Hotel Ratings as we thought it was more meaningful for the users but the product itself was named Hotel Sentiments...

File: README.md [hunk 3]
```diff
@@ -307,6 +307,8 @@ HotelOffer[] offers = amadeus.shopping.hotelOffers.get(Params
 HotelOffer hotelOffer = amadeus.shopping.hotelOffersByHotel.get(Params.with("hotelId", "BGLONBGB"));
 // Confirm the availability of a specific offer
 HotelOffer offer = amadeus.shopping.hotelOffer("4BA070CE929E135B3268A9F2D0C51E9D4A6CF318BA10485322FA2C7E78C7852E").get();
+// Hotel Ratings
+HotelRating[] hotelSentiments = amadeus.ereputation.hotelSentiments.get(Params.with("hotelIds", "ELONMFS,ADNYCCTB"));
```
Comment (anthonyroux) [inline README.md]: HotelSentiment[] hotelSentiments

File: src/main/java/com/amadeus/resources/HotelSentiment.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+public class HotelSentiment extends Resource {
+  protected HotelSentiment() {}
+
+  private @Getter String hotelId;
+  private @Getter String type;
+  private @Getter int overallRating;
+  private @Getter int numberOfReviews;
+  private @Getter Sentiment sentiments;
+
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/HotelSentiment.java]: tag `@ToString` missing

File: src/main/java/com/amadeus/resources/HotelSentiment.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package com.amadeus.resources;
+
+import lombok.Getter;
+
+public class HotelSentiment extends Resource {
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/HotelSentiment.java]: tag `@ToString` missing

File: src/main/java/com/amadeus/resources/HotelSentiment.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.amadeus.resources;
+
+import lombok.Getter;
+
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/resources/HotelSentiment.java]:
```
/**
 * An HotelSentiment object as returned by the Hotel Ratings API.
 * @see HotelSentiment#get()
 */
```

File: src/main/java/com/amadeus/ereputation/HotelSentiments.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package com.amadeus.ereputation;
+
+import com.amadeus.Amadeus;
+import com.amadeus.Params;
+import com.amadeus.Response;
+import com.amadeus.exceptions.ResponseException;
+import com.amadeus.resources.HotelSentiment;
+import com.amadeus.resources.Resource;
+
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/ereputation/HotelSentiments.java]:
```
/**
 * <p>
 *   A namespaced client for the
 *   <code>/v2/e-reputation/hotel-sentiments</code> endpoints.
 * </p>
 *
 * <p>
 *   Access via the Amadeus client object.
 * </p>
 *
 * <pre>
 * Amadeus amadeus = Amadeus.builder("clientId", "secret").build();
 * amadeus.ereputation.hotelSentiments;</pre>
 */
```

File: src/main/java/com/amadeus/ereputation/HotelSentiments.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+public class HotelSentiments {
+  private Amadeus client;
+
+  public HotelSentiments(Amadeus client) {
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/ereputation/HotelSentiments.java]:
```
  /**
   * Constructor.
   * @hide
   */
```

File: src/main/java/com/amadeus/ereputation/HotelSentiments.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    this.client = client;
+  }
+
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/ereputation/HotelSentiments.java]:
```
  /**
   * <p>
   *   Hotel ratings based on automated sentiment analysis algorithm
   *    applied on the online reviews.
   * </p>
   *
   * <pre>
   * amadeus.ereputation.hotelSentiments.get(Params
   *   .with("hotelIds", "ELONMFS,ADNYCCTB"));</pre>
   *
   * @param params the parameters to send to the API
   * @return an API response object
   * @throws ResponseException when an exception occurs
   */
```

File: src/main/java/com/amadeus/ereputation/HotelSentiments.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  public HotelSentiment[] get(Params params) throws ResponseException {
+    Response response = client.get("/v2/e-reputation/hotel-sentiments", params);
+    return (HotelSentiment[]) Resource.fromArray(response, HotelSentiment[].class);
+  }
```
Comment (anthonyroux) [inline src/main/java/com/amadeus/ereputation/HotelSentiments.java]:
```
  /**
   * Convenience method for calling <code>get</code> without any parameters.
   * @see HotelSentiment#get()
   */
  public HotelSentiment[] get() throws ResponseException {
    return get(null);
  }
```

Comment (SealSoft) [reply src/main/java/com/amadeus/ereputation/HotelSentiments.java]: The above method has mandatory parameters so i ignored the common `get` without params method in order to be less confusing for the developer

**Pull request merged**